storage/series/policy.go

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package series
+
+import (
+	"errors"
+	"fmt"
+)
+
+const (
+	// DefaultCachePolicy is the default cache policy.
+	DefaultCachePolicy = CacheRecentlyRead
+)
+
+var (
+	errCachePolicyUnspecified = errors.New("series cache policy unspecified")
+)
+
+// CachePolicy is the series cache policy.
+type CachePolicy uint
+
+const (
+	// CacheAll specifies that all series must be cached at all times
+	// which requires loading all into cache on bootstrap and never
+	// expiring series from memory until expired from retention.
+	CacheAll CachePolicy = iota
+	// CacheAllMetadata specifies that all series metadata but not the
+	// data itself must be cached at all times and the metadata is never
+	// expired from memory until expired from retention.
+	// TODO: Remove this once recently read is production grade.
+	CacheAllMetadata
+	// CacheRecentlyRead specifies that series that are recently read
+	// must be cached, configurable by the namespace block expiry after
+	// not accessed period.
+	CacheRecentlyRead
+)
+
+// ValidCachePolicies returns the valid series cache policies.
+func ValidCachePolicies() []CachePolicy {
```

Comment:
super nit: In the session file we just have a global slice. Would be nice to be consistent in terms of having a function that returns it, or a global slice

Reply:
I use a function to make it impossible for caller to mutate it.  I can update sessions usage, although its only internal to the package there..